main.go

```diff
@@ -52,17 +52,24 @@ func main() {
 		}
 	}
 
-	{
-		osSignals := make(chan os.Signal, 1)
-		signal.Notify(osSignals, os.Interrupt, syscall.SIGTERM)
-		<-osSignals
-		// stop the program
-		// sometimes after the program exits on Windows, the ports are still occupied and "listening".
-		// so manually closes these listeners when the program exits.
-		for _, listener := range service.ListenerArray {
-			if listener != nil { // avoid null pointers
-				listener.Close()
+	osSignals := make(chan os.Signal, 1)
+	signal.Notify(osSignals, os.Interrupt, os.Kill, syscall.SIGTERM, syscall.SIGHUP)
```

Comment:
os.Kill seems like a uncatchable signal and I have removed it in previous commit. Please check it again.